### ISSUE TITLE
K8S: Shells/Logs Close Randomly after 1.2 upgrade

### ISSUE THREAD

**Issue by rezonant** [rancher/rancher#6913]
**Rancher Version:**
1.2.0

**Docker Version:**
1.12.3

**OS and where are the hosts located? (cloud, bare metal, etc):**

Ubuntu 16.04.1 LTS (with KVM) (4.4.0) / Digital Ocean

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node, internal DB via container

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes 1.4.6

**Steps to Reproduce:**
Randomly, shells and logs will either fail to connect (hanging at Connecting or showing Disconnected immediately), or close unexpectedly while being used / read, though the containers being connected to are still running. Happens with the Kubectl UI as well, sometimes immediately, to the point where I cannot even type commands into it. Happens on the k8s environment I upgraded after moving to 1.2.0 as well as a brand new environment. Notably, it comes in waves, where the problem gets bad, and then later it goes away and I'm able to execute things without as many failures.

**Logs**

The Rancher container show the following logs if I open the Kubectl UI page and wait for it to close itself (in this case it took about 15 seconds):

```
time="2016-12-05T01:36:55Z" level=info msg="Handling backend connection request."
time="2016-12-05T01:36:55Z" level=info msg="Registering backend for host 0b3e09bc-8403-4a22-8878-a0bc4239a9d8 with session ID bbab5088-e5c4-4b50-8848-88708a5466be."
time="2016-12-05T01:36:56Z" level=info msg="Handling backend connection request."
time="2016-12-05T01:36:56Z" level=info msg="Registering backend for host f1cb5ff6-8710-454b-8eee-dc321a229bcf with session ID 37538923-3982-43ad-a018-cb096f4090ea."
time="2016-12-05T01:37:02Z" level=info msg="Shutting down backend 0b3e09bc-8403-4a22-8878-a0bc4239a9d8. Connection closed because: websocket: close 1006 unexpected EOF."
time="2016-12-05T01:37:02Z" level=info msg="Removed backend. Key: 0b3e09bc-8403-4a22-8878-a0bc4239a9d8. Session ID bbab5088-e5c4-4b50-8848-88708a5466be ."
time="2016-12-05T01:37:03Z" level=info msg="Shutting down backend f1cb5ff6-8710-454b-8eee-dc321a229bcf. Connection closed because: websocket: close 1006 unexpected EOF."
time="2016-12-05T01:37:03Z" level=info msg="Removed backend. Key: f1cb5ff6-8710-454b-8eee-dc321a229bcf. Session ID 37538923-3982-43ad-a018-cb096f4090ea ."
time="2016-12-05T01:37:10Z" level=info msg="Handling backend connection request."
time="2016-12-05T01:37:10Z" level=info msg="Registering backend for host 0b3e09bc-8403-4a22-8878-a0bc4239a9d8 with session ID 91b197b2-2614-47cc-8145-dc8609d689d5."
time="2016-12-05T01:37:12Z" level=info msg="Handling backend connection request."
time="2016-12-05T01:37:12Z" level=info msg="Registering backend for host f1cb5ff6-8710-454b-8eee-dc321a229bcf with session ID ebe2525d-4499-4fc6-a494-504093ff9bc2."
time="2016-12-05T01:37:17Z" level=info msg="Shutting down backend 0b3e09bc-8403-4a22-8878-a0bc4239a9d8. Connection closed because: websocket: close 1006 unexpected EOF."
time="2016-12-05T01:37:17Z" level=info msg="Removed backend. Key: 0b3e09bc-8403-4a22-8878-a0bc4239a9d8. Session ID 91b197b2-2614-47cc-8145-dc8609d689d5 ."
time="2016-12-05T01:37:17Z" level=info msg="Shutting down backend 9578ccf2-4e1d-4d1e-9e90-2abc1a158e2d. Connection closed because: websocket: close 1006 unexpected EOF."
time="2016-12-05T01:37:17Z" level=info msg="Removed backend. Key: 9578ccf2-4e1d-4d1e-9e90-2abc1a158e2d. Session ID b7462911-d63a-46c4-aa92-24a53f1804ce ."
time="2016-12-05T01:37:17Z" level=info msg="Shutting down backend f1cb5ff6-8710-454b-8eee-dc321a229bcf. Connection closed because: websocket: close 1006 unexpected EOF."
time="2016-12-05T01:37:17Z" level=info msg="Removed backend. Key: f1cb5ff6-8710-454b-8eee-dc321a229bcf. Session ID ebe2525d-4499-4fc6-a494-504093ff9bc2 ."
time="2016-12-05T01:37:17Z" level=info msg="Shutting down backend 1fd111b6-184b-4d79-a674-d53cb719a051. Connection closed because: websocket: close 1006 unexpected EOF."
time="2016-12-05T01:37:17Z" level=info msg="Removed backend. Key: 1fd111b6-184b-4d79-a674-d53cb719a051. Session ID 017a48e2-a7d6-4be6-b57c-5420d98a3224 ."
```

I also see this error occasionally from the Rancher container log:
```
2016-12-05 01:42:49,686 ERROR [:] [] [] [] [TaskScheduler-2] [i.c.p.a.s.ping.impl.PingMonitorImpl ] Failed to get ping from agent [124] count [3]
```

Notably, there was a single browser window open on this Rancher instance, and I was doing nothing, so it seems these websocket disconnects are happening on their own. They are happening even when there's no visible problem in the Rancher UI

I'm also seeing similar failures in the browser console:
```
VM154:35 WebSocket connection to 'wss://rancher.example.com/v1/exec/?token=eyJhbGciO..27Gw' failed: HTTP Authentication failed; no valid credentials available
```
This error is printed whenever I attempt to connect to Kubectl but it never succeeds (ie hangs at Connecting...). If I am able to get Kubectl running, but it fails afterwards, I can see that the Websockets connection indeed gets disconnected (during this last test it stayed open for some 2.8 seconds according to Chrome Inspector).

That Websockets connection only had 2 frames, the last one was 
`PiA=` which is base64 for '> ' which is the prompt of kubectl. So it seems to just drop the connection randomly.

**Expected:**
Terminals and logs should connect properly and stay connected

**Comment by rezonant** [comment]
I did two things and this problem disappeared. (A) I deactivated and ultimately removed my original problematic upgraded Kubernetes environment as the new one was put together with the new Stacks system anyway and everything was moved over.
And (B) I noticed a set of containers as "Standalone" containers on the Hosts screen which had been spinning probably the entire time I was testing. These would have been attempting to access NFS, and may  have been disrupted if NFS failed. These containers were part of a stack deployed from catalog, so I don't think they should've been marked Standalone. I removed these.

Either way, I think the problem has subsided and perhaps the problem was the upgraded environment, or the Kubernetes pods stuck after failed NFS mounts, will close this and reopen if if the specific problem continues on this cluster.
